src/initializeCommunicationDir.ts

```diff
@@ -0,0 +1,30 @@
+import { statSync, mkdirSync } from "fs";
+import { S_IWOTH } from "constants";
+import { getuid } from "process";
+import {
+  getRequestPath,
+  getResponsePath,
+  getCommunicationDirPath
+} from "./paths";
+import { unlinkIfExistsSync } from "./fileUtils";
+
+export function initializeCommunicationDir() {
+  const communicationDirPath = getCommunicationDirPath();
+
+  console.log(`Creating communication dir ${communicationDirPath}`);
+  mkdirSync(communicationDirPath, { recursive: true, mode: 0o770 });
+
+  const stats = statSync(communicationDirPath);
```

Comment:
This needs to be `lstat`, not `stat` - stat follows symlinks, which allows another user on the same machine to symlink to a directory owned by the target user during startup, and edit the symlink later to point wherever they want.